compiler/utils.go

```diff
@@ -643,3 +645,38 @@ func endsWithReturn(stmts []ast.Stmt) bool {
 func encodeIdent(name string) string {
 	return strings.Replace(url.QueryEscape(name), "%", "$", -1)
 }
+
+// formatJSStructTagVal returns a string of Javascript code appropriate for
+// accessing the property identified by jsTag. If the jsTag value can be safely
+// encoded in Javascript using the dot notation this is used, else we fallback
+// to the bracket notation. For more details see:
+//
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_Accessors
+//
+// Uses definition of an identifier from
+// https://developer.mozilla.org/en-US/docs/Glossary/Identifier
+func formatJSStructTagVal(jsTag string) string {
+	useDot := true
```

Comment:
I'm not sure I agree. We want the default behaviour to be to use the dot notation because, as noted before, this is more efficient in output terms. "Dot notation" is taken directly from the MDN. To my mind it's better to have the variable name reflect the intended semantics, hence `useDot`.

Reply:
> We want the default behaviour to be to use the dot notation because, as noted before, this is more efficient in output terms.

My suggestion doesn't contradict that. `needsEscaping` controls whether non-default behavior of using escaping is used. If its value remains `false`, the default behavior (using dot notation) is used.

It's similar to when you handle error conditions (non-default behaviors), you typically write:

```Go
f, err := os.Open("file.txt")
if err != nil {
    // less common (non-default) path
    return fmt.Errorf("couldn't open file.txt: %v", err)
}
// normal (default) path
```

Instead of:

```Go
successfullyOpened := true
f, err := os.Open("file.txt")
if err != nil {
    successfullyOpened = false
}
if successfullyOpened {
    // normal (default) path
}
// less common (non-default) path
return fmt.Errorf("couldn't open file.txt: %v", err)
```

Reply:
> My suggestion doesn't contradict that. needsEscaping controls whether non-default behavior of using escaping is used. If its value remains false, the default behavior (using dot notation) is used.

No I understand what you were getting at, it's just the name `needsEscaping` that doesn't make sense to me. I'm quite happy having the inverse logic, but if we were to do so then `useBracket` would probably be a better name (it's the [alternative form](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_accessors)).